File: Core Data/WWDC 2015/Whats_new_in_core_data/iPhoneCoreDataRecipes/Classes/TemperatureConverterViewController.m
/*
 Copyright (C) 2017 Apple Inc. All Rights Reserved.
 See LICENSE.txt for this sample’s licensing information
 
 Abstract:
 View controller to display cooking temperatures in Centigrade, Fahrenheit, and Gas Mark.
 */

#import "TemperatureConverterViewController.h"
#import "TemperatureCell.h"

@interface TemperatureConverterViewController ()

@property (nonatomic, strong) NSArray *temperatureData;

@end


#pragma mark -

@implementation TemperatureConverterViewController

static NSString *MyIdentifier = @"MyIdentifier";


#pragma mark - UITableViewDataSource

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    
    return self.temperatureData.count;
}

- (UITableViewCell *)tableView:(UITableView *)aTableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    
    // Create a new TemperatureCell.
    TemperatureCell *cell =
        (TemperatureCell *)[aTableView dequeueReusableCellWithIdentifier:MyIdentifier
                                                            forIndexPath:indexPath];
    
    // Configure the temperature cell with the relevant data.
    NSDictionary *temperatureDictionary = (self.temperatureData)[indexPath.row];
    [cell setTemperatureDataFromDictionary:temperatureDictionary];
    
    return cell;
}


#pragma mark - Temperature data

- (NSArray *)temperatureData {
	
	if (_temperatureData == nil) {
		// Get the temperature data from the TemperatureData property list.
		NSString *temperatureDataPath = [[NSBundle mainBundle] pathForResource:@"TemperatureData" ofType:@"plist"];
		NSArray *array = [[NSArray alloc] initWithContentsOfFile:temperatureDataPath];
		self.temperatureData = array;
	}
	return _temperatureData;
}


#pragma mark - Memory management

- (void)didReceiveMemoryWarning {
    
	[super didReceiveMemoryWarning];
	self.temperatureData = nil;
}

@end

File: Core Data/Core Data Sample Codes/iPhoneCoreDataRecipes/Recipes_Swift/Recipes_Swift/Converter View Controllers/TemperatureConverterViewController.swift
/*
  TemperatureConverterViewController.swift
  Recipes_Swift

  Created by Seyed Samad Gholamzadeh on 2/28/18.
  Copyright © 2018 Seyed Samad Gholamzadeh. All rights reserved.

Abstract:
View controller to display cooking temperatures in Centigrade, Fahrenheit, and Gas Mark.
*/


import UIKit

class TemperatureConverterViewController: UITableViewController {

	var temperatureData: [Any]! = {
		// Get the temperature data from the TemperatureData property list.
		let temperatureDataPath = Bundle.main.url(forResource: "TemperatureData", withExtension: "plist")
		let array = NSArray(contentsOf: temperatureDataPath!)
		return array as! [Any]
	}()
	
	let MyIdentifier = "MyIdentifier"
	
    override func viewDidLoad() {
        super.viewDidLoad()

        // Uncomment the following line to preserve selection between presentations
        // self.clearsSelectionOnViewWillAppear = false

        // Uncomment the following line to display an Edit button in the navigation bar for this view controller.
        // self.navigationItem.rightBarButtonItem = self.editButtonItem
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
		self.temperatureData = nil
    }

    // MARK: - Table view data source

    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        // #warning Incomplete implementation, return the number of rows
        return self.temperatureData.count
    }

    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: MyIdentifier, for: indexPath) as! TemperatureCell

		// Configure the temperature cell with the relevant data.
		let temperatureDictionary = self.temperatureData[indexPath.row] as! [String: String]
		cell.setTemperatureDataFrom(dictionary: temperatureDictionary)


        return cell
    }

    /*
    // Override to support conditional editing of the table view.
    override func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
        // Return false if you do not want the specified item to be editable.
        return true
    }
    */

    /*
    // Override to support editing the table view.
    override func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCellEditingStyle, forRowAt indexPath: IndexPath) {
        if editingStyle == .delete {
            // Delete the row from the data source
            tableView.deleteRows(at: [indexPath], with: .fade)
        } else if editingStyle == .insert {
            // Create a new instance of the appropriate class, insert it into the array, and add a new row to the table view
        }    
    }
    */

    /*
    // Override to support rearranging the table view.
    override func tableView(_ tableView: UITableView, moveRowAt fromIndexPath: IndexPath, to: IndexPath) {

    }
    */

    /*
    // Override to support conditional rearranging of the table view.
    override func tableView(_ tableView: UITableView, canMoveRowAt indexPath: IndexPath) -> Bool {
        // Return false if you do not want the item to be re-orderable.
        return true
    }
    */

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
